Add role based auth to CloudWatch exporter doc

minkimipt opened the pull request:
Role based authentication is added for CloudWatch exporter that allows to authenticate with users' AWS accounts and we need to document configuration that needs to be done on the user side to make it work.

# Description

[Short summary of why you created this PR]

# Links

Fixes https://github.com/timescale/docs/issues/3155

# Writing help

For information about style and word usage, see the [style guide](https://docs.timescale.com/about/latest/contribute-to-docs/)

# Review checklists

Reviewers: use this section to ensure you have checked everything before approving this PR:

## Subject matter expert (SME) review checklist

*   [ ] Is the content technically accurate?
*   [ ] Is the content complete?
*   [ ] Is the content presented in a logical order?
*   [ ] Does the content use appropriate names for features and products?
*   [ ] Does the content provide relevant links to further information?

## Documentation team review checklist

*   [ ] Is the content free from typos?
*   [ ] Does the content use plain English?
*   [ ] Does the content contain clear sections for concepts, tasks, and references?
*   [ ] Have any images been uploaded to the correct location, and are resolvable?
*   [ ] If the page index was updated, are redirects required
      and have they been implemented?
*   [ ] Have you checked the built version of this content?

minkimipt committed:
Add role based auth to CloudWatch exporter doc

Role based authentication is added for CloudWatch exporter that allows
to authenticate with users' AWS accounts and we need to document
configuration that needs to be done on the user side to make it work.

niconosenzo committed:
add provider URLs and role example

niconosenzo committed:
Clarify what the URL serves for

billy-the-fish committed:
chore: first go updating the configuration procedure.

minkimipt commented on use-timescale/metrics-logging/integrations.md:
````diff
@@ -104,9 +104,87 @@ this -->
+    } 
+```
+
+Trust Policy:
````
@niconosenzo why do we call it trust policy here if it's a role?

niconosenzo replied:
🤔 it was the trust policy that began in line 152 of my commit. Will double check though

minkimipt replied:
I think if we want to be aligned with how AWS calls them in [the blog](https://aws.amazon.com/blogs/containers/cross-account-iam-roles-for-kubernetes-service-accounts/), we need to call it a role with a trust policy

```
3. Create a role in the shared_content account that provides ReadOnlyAccess to all objects in pics bucket. Users federated by OIDC provider are allowed to assume this role.

 "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Principal": {
        "Federated": "arn:aws:iam::shared_content_account_id:oidc-provider/oidc.eks.us-east-1.amazonaws.com/id/oidc-id"
      },
      "Action": "sts:AssumeRoleWithWebIdentity",
      "Condition": {
        "StringEquals": {
          "oidc.eks.us-east-1.amazonaws.com/id/oidc-id:aud": "sts.amazonaws.com"
        }
      }
    }
  ]
}
```

niconosenzo replied:
TBH, for me it's always easier to split a role among permission policies and trust policies. But yeah we can call it a role if that makes more sense.

billy-the-fish committed:
chore: updates after first review.

billy-the-fish committed:
chore: updates after first review.

minkimipt commented on use-timescale/metrics-logging/integrations.md:
```diff
@@ -65,40 +65,129 @@ alt="Screenshot of the menu for adding a Datadog exporter" />
+    AWS keys give access to your AWS services. To keep your AWS account secure,
+    restrict users to the minimum required permissions. Always store your keys in a
+    safe location. To avoid this issue, use the IAM role authentication method. 
+    </Highlight>
```
@billy-the-fish if we move the warning `Highlight` inside the `Procedure`, I think it would actually start displaying properly only in conjunction with `CloudWatch credentials` where it should belong
```suggestion

    <Procedure>

     When you use CloudWatch credentials, you link an Identity and Access Management (IAM) 
     user with access to Cloudwatch only with your Timescale Service:  

    1. Retrieve the user information from [IAM > Users in AWS console][list-iam-users].   
    
       If you do not have an AWS user with access restricted to CloudWatch only, 
       [create one][create-an-iam-user]. 
       For more information, see [Creating IAM users (console)][aws-access-keys].
    2. Enter the credentials for the AWS IAM user. 

    <Highlight type="warning">
    AWS keys give access to your AWS services. To keep your AWS account secure,
    restrict users to the minimum required permissions. Always store your keys in a
    safe location. To avoid this issue, use the IAM role authentication method. 
    </Highlight>
    
   </Procedure>
```

billy-the-fish committed:
Update use-timescale/metrics-logging/integrations.md

Nice.

Co-authored-by: Danil Zhigalin <danil@timescale.com>
Signed-off-by: Iain Cox <iain@timescale.com>

billy-the-fish committed:
fix: update metatags for MDX.

minkimipt commented:
@billy-the-fish last docs build that I saw looked good to me. The PR was actually started by me, that's why I'm not allowed to approve it, but I think it looks good. @niconosenzo do you want to have a final look to?

minkimipt committed:
Replace trust policy to a role

AWS docs refer to this as a role. Make sure we are talking about the
same thing in our docs.

niconosenzo commented on use-timescale/metrics-logging/integrations.md:
```diff
@@ -12,21 +12,28 @@ cloud_ui:
+the lifecycle of data exporters.
 
-Export telemetry data by:
+## Create a data exporter and securely connect it a monitoring tool
```
```suggestion
## Create a data exporter and securely connect it to a monitoring tool
```

niconosenzo reviewed (approved):
Thanks!

billy-the-fish committed:
Update use-timescale/metrics-logging/integrations.md

Co-authored-by: Nicolas Nosenzo <niconosenzo@gmail.com>
Signed-off-by: Iain Cox <iain@timescale.com>

billy-the-fish committed:
Merge branch 'latest' into update-cloudwatch-exporter-ingetrations

billy-the-fish committed:
Merge branch 'latest' into update-cloudwatch-exporter-ingetrations

billy-the-fish reviewed (approved):
LGTM
